server/routes/auth.js

```diff
@@ -0,0 +1,121 @@
+const express = require("express");
+const { check, validationResult, cookie } = require("express-validator");
+const UserModel = require("../models/userModel");
+const jwt = require("jsonwebtoken");
+const router = express.Router();
+
+const setJwt = (user) => {
+  const token = jwt.sign({ id: user.id, email: user.email }, process.env.JWT_SECRET, {
+    expiresIn: "14d",
+  });
+
+  return token;
+};
+
+router.post(
+  "/register",
+  [
+    check("name", "Name must not be empty").notEmpty(),
+    check("password", "Must be at least 6 characters long").isLength({
+      min: 6,
+    }),
+    check("email", "Must be an Email").isEmail().notEmpty(),
+  ],
+  async (req, res) => {
+    // if (req.cookies.token) res.redirect('req.headers.host')
+
+    const errors = validationResult(req);
```

Comment:
It's cleaner to move the validation logic into a separate function (a validation middleware). That way, all your validation logic is in one place and easy to locate, read and reuse if you need to change/expand it in the future.